lib/document.js

```diff
@@ -666,16 +606,88 @@ module.exports = class extends Fragment {
       throw new TypeError('ExternalDocument does not have page ' + page)
     }
 
-    this._begin(null)
-    this._pending.push(() => external.write(this._doc, page))
+    this._rootFragment._begin(null)
+    this._pending.push(() => external.write(this, page))
   }
 
   setTemplate(external) {
     if (!(external instanceof ExternalDocument)) {
       throw new TypeError('argument must be of type ExternalDocument')
     }
 
-    this._begin(null)
-    this._pending.push(() => external.setAsTemplate(this._doc))
+    this._rootFragment._begin(null)
+    this._pending.push(() => external.setAsTemplate(this))
+  }
+
+  text(text, opts) {
+    return this._rootFragment.text(text, opts)
```

Comment:
Forwarding the methods would look like `Fragment.prototype.text.call(this, text, opts)` if we can remove the `_rootFragment` as mentioned above